src/utils/wallet.ts

```diff
@@ -0,0 +1,225 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+
+import { crypto, HDPublicKey, HDPrivateKey, Address } from 'bitcore-lib';
+import Mnemonic from 'bitcore-mnemonic';
+import { HD_WALLET_ENTROPY } from '../constants';
+import { XPubError, InvalidWords, UncompressedPubKeyError } from '../errors';
+import Network from '../models/network';
+import _ from 'lodash';
+
+
+const wallet = {
+  /**
+   * Verify if words passed to generate wallet are valid. In case of invalid, returns message
+   *
+   * @param {string} words Words (separated by space) to generate the HD Wallet seed
+   *
+   * @return {Object} {'valid': boolean, 'invalidWords': Array[string], 'words': string} where 'words' is a cleaned
+   * string with the words separated by a single space and invalidWords is an array of invalid words
+   * @throws {InvalidWords} In case the words string is invalid
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  wordsValid(words: string): {valid: boolean, invalidWords?: string[], words?: string} {
+    let newWordsString = '';
+    if (_.isString(words)) {
+      // 1. Replace all non ascii chars by a single space
+      // 2. Remove one or more spaces (or line breaks) before and after the 24 words
+      // 3. Set text to lower case
+      newWordsString = words.replace(/[^A-Za-z0-9]/g, ' ').replace(/\s+/g, ' ').trim().toLowerCase();
+      const wordsArray = newWordsString.split(' ');
+      if (wordsArray.length !== 24) {
+        // Must have 24 words
+        throw new InvalidWords('Must have 24 words.')
```

Comment:
Shouldn't we validate the given words and return which ones are invalid? It seems useful when user is still typing the words and we'd like to advance potential errors.

Reply:
Can you see this commit? https://github.com/HathorNetwork/hathor-wallet-lib/pull/110/commits/e15d98df6aa37801ae854384cf60ecc451e8ca98

I've done this change and the one below about throwing the error.

Reply:
That's great! Thanks!